examples/plot_sampling_target_usage.py

```diff
@@ -0,0 +1,241 @@
+           autopct='%1.1f%%')
+    ax.axis('equal')
+
+
+###############################################################################
+# Creation of an imbalanced data set from a balanced data set
+###############################################################################
+
+###############################################################################
+# We will show how to use the parameter ``sampling_target`` when dealing with
+# the ``make_imbalance`` function. For this function, this parameter accepts
+# both dictionary and callable. When using a dictionary, each key will
+# correspond to the class of interest and the corresponding value will be the
+# number of samples desired in this class.
+
+iris = load_iris()
+
+print('Information of the original iris data set: \n {}'.format(
+    Counter(iris.target)))
+plot_pie(iris.target)
+
+sampling_target = {0: 10, 1: 20, 2: 30}
+X, y = make_imbalance(iris.data, iris.target, sampling_target=sampling_target)
+
+print('Information of the iris data set after making it'
+      ' imbalanced using a dict: \n sampling_target={} \n y: {}'
+      .format(sampling_target, Counter(y)))
+plot_pie(y)
+
+###############################################################################
+# You might required more flexibility and require your own heuristic to
+# determine the number of samples by class and you can define your own callable
+# as follow. In this case we will define a function which will use a float
+# multiplier to define the number of samples per class.
+
+
+def ratio_multiplier(y):
+    multiplier = {0: 0.5, 1: 0.7, 2: 0.95}
+    target_stats = Counter(y)
+    for key, value in target_stats.items():
+        target_stats[key] = int(value * multiplier[key])
+    return target_stats
+
+
+X, y = make_imbalance(iris.data, iris.target, sampling_target=ratio_multiplier)
+
+print('Information of the iris data set after making it'
+      ' imbalanced using a callable: \n sampling_target={} \n y: {}'
+      .format(sampling_target, Counter(y)))
+plot_pie(y)
+
+###############################################################################
+# Using ``sampling_target`` in resampling algorithm
+###############################################################################
+
+###############################################################################
+# ``sampling_target`` has a ``float``
+# ...................................
+#
+# ``sampling_target`` can be given a ``float``. For **under-sampling
+# methods**, it corresponds to the ratio :math:`\\alpha_{us}` defined by
+# :math:`N_{rM} = \\alpha_{us} \\times N_{m}` where :math:`N_{rM}` and
+# :math:`N_{m}` are the number of samples in the majority class after
+# resampling and the number of samples in the minority class, respectively.
+
+# select only 2 classes since the ratio make sense in this case
+binary_mask = np.bitwise_or(y == 0, y == 2)
+binary_y = y[binary_mask]
+binary_X = X[binary_mask]
+
```

Comment:
can you show the counter of the data? So you can afterwards compare the number after resampling